src/relay.rs

```diff
@@ -151,6 +144,10 @@ pub struct Network<T: DuplexTransport> {
     free: bool,
     confirmations: u64,
     anchor_frequency: u64,
+    chain_id: u64,
+    keydir: String,
+    password: String,
+    nonce: Arc<Mutex<U256>>,
```

Comment:
Think we could probably get away with an atomic int here, debatable though.

Reply:
I tried using AtomicU64 but it's experimental and the compiler didn't like it

Reply:
Yeah for some reason the atomic fixed sized ints have been nighly only for forever, AtomicUsize would probably be fine but a bit arch dependant, this also works just is probably less efficient

Reply:
@mjkoo updated with 7b26efa